lib/private-libwebsockets.h

```diff
@@ -39,6 +39,8 @@
 #endif
 
 #if defined(WIN32) || defined(_WIN32)
+#define WINVER 0x0601
```

Comment:
When left out, I get compiler warnings and linker errors due to absense of getnameinfo(), getaddrinfo(), and freeaddrinfo(). The reason for that is an #if (_WIN32_WINNT >= 0x0501) in the Ws2tcpip.h header included in the MinGW distribution. Given that, I agree that we could be less restrictive by going back to #define _WIN32_WINNT 0x0501. Nevertheless, MSDN tells that the minimum supported client for getnameinfo() is Windows 8.1, Windows Vista:

https://msdn.microsoft.com/en-us/library/windows/desktop/ms738532(v=vs.85).aspx